### PR TITLE
nixos/lxd: add productionSetup option

### PR DESCRIPTION
**Pull request by mt-caret**
###### Motivation for this change

Adds an option to fix errors that may come up for ZFS storage backend users.
See <https://discuss.linuxcontainers.org/t/weird-error-for-zfs-storage-backend/6008/3> for details.

###### Things done

Tested on own machine.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/virtualisation/lxd.nix [hunk 1]
```diff
@@ -69,8 +81,9 @@ in
         ExecStart = "@${pkgs.lxd.bin}/bin/lxd lxd --group lxd";
         Type = "simple";
         KillMode = "process"; # when stopping, leave the containers alone
+        LimitMEMLOCK = mkIf cfg.productionSetup "infinity";
+        LimitNOFILE = mkIf cfg.productionSetup "1048576";
```
Comment (megheaiulian) [inline nixos/modules/virtualisation/lxd.nix]: I think this service settings should always be added (not in production only). The debian lxd packages seem to do that too https://github.com/lxc/lxd-pkg-ubuntu/blob/dpm-bionic/debian/lxd.service#L15.

Comment (mt-caret) [reply nixos/modules/virtualisation/lxd.nix]: Good point. I've added some stuff from https://github.com/lxc/lxd-pkg-ubuntu/blob/dpm-xenial/debian/lxd.service , how does this look?

Comment (megheaiulian) [reply nixos/modules/virtualisation/lxd.nix]: Looks great! I would only rename the option from `productionSetup` to `recommendedSysctlSettings` in the style of the `recommended*` nginx options in nixos: https://nixos.org/nixos/options.html#nginx.reco

**Review by megheaiulian — APPROVED** [review]

File: nixos/modules/virtualisation/lxd.nix (hunk 1 above)
Comment (mt-caret) [reply nixos/modules/virtualisation/lxd.nix]: Fixed.

**Comment by megheaiulian** [comment]
@mt-caret I am not sure who we should be ping to review and merge this ?

**Comment by mt-caret** [comment]
Possibly @wucke13 @Lassulus from https://github.com/NixOS/nixpkgs/pull/70209 ?

**Review by Lassulus — APPROVED** [review]
LGTM

**Pull request merged**